diff --git a/node_modules/@keystatic/core/dist/keystatic-core-ui.js b/node_modules/@keystatic/core/dist/keystatic-core-ui.js
index 24d4713..54216f0 100644
--- a/node_modules/@keystatic/core/dist/keystatic-core-ui.js
+++ b/node_modules/@keystatic/core/dist/keystatic-core-ui.js
@@ -2019,7 +2019,7 @@ function useUpsertItem(args) {
             }
             const tree = scopeEntriesWithPathPrefix(await fetchGitHubTreeData(refData.data.repository.ref.target.oid, args.config), args.config);
             const treeKey = getTreeKey(getDirectoriesForTreeKey(object(args.schema), args.basePath, (_args$slug3 = args.slug) === null || _args$slug3 === void 0 ? void 0 : _args$slug3.value, args.format), tree.tree);
-            if (treeKey === args.currentLocalTreeKey) {
+            if (treeKey === (args.currentLocalTreeKey !== undefined ? args.currentLocalTreeKey : getTreeKey(getDirectoriesForTreeKey(object(args.schema), args.basePath, (_args$slug3 = args.slug) === null || _args$slug3 === void 0 ? void 0 : _args$slug3.value, args.format), new Map()))) {
               result = await runMutation(refData.data.repository.ref.target.oid);
             } else {
               setState({
